cylc/flow/tests/test_config.py

```diff
@@ -211,3 +212,32 @@ def test_family_inheritance_and_quotes(self):
                         ['MAINFAM_major1_minor10'])
                 assert 'goodbye_0_major1_minor10' in \
                        config.runtime['descendants']['SOMEFAM']
+
+
+def test_queue_config(caplog):
+    suiterc_content = """
+[scheduling]
+   [[queues]]
+       [[[q1]]]
+           members = A, B
+       [[[q2]]]
+           members = x
+   [[dependencies]]
+       graph = "x => y"
+[runtime]
+   [[A]]
+   [[B]]
+   [[x]]
+       inherit = A, B
+   [[y]]
+    """
+    workflow_directory = Path(mkdtemp())
+    suite_rc = Path(workflow_directory, "suite.rc")
+    with suite_rc.open(mode="w") as f:
+        f.write(suiterc_content)
+        f.flush()
+    config = SuiteConfig(suite="qtest", fpath=f.name)
+    config.configure_queues()
+    log = caplog.messages[0].split('\n')
+    assert log[0] == "Queue configuration warnings:"
+    assert log[1] == "+ q2: ignoring x (already assigned to a queue)"
```

Comment:
Instead of `mkdtemp`, we can also use `pytest`'s [`tmpdir`](https://docs.pytest.org/en/latest/tmpdir.html).

This one worked running the test locally in the IDE.

```diff
diff --git a/cylc/flow/tests/test_config.py b/cylc/flow/tests/test_config.py
index 251c4fe5a..899e50585 100644
--- a/cylc/flow/tests/test_config.py
+++ b/cylc/flow/tests/test_config.py
@@ -16,7 +16,7 @@
 
 import os
 import pytest
-from tempfile import TemporaryDirectory, NamedTemporaryFile, mkdtemp
+from tempfile import TemporaryDirectory, NamedTemporaryFile
 from pathlib import Path
 from cylc.flow.config import SuiteConfig
 from cylc.flow.tests.util import CylcWorkflowTestCase
@@ -214,7 +214,7 @@ class TestSuiteConfig(object):
                        config.runtime['descendants']['SOMEFAM']
 
 
-def test_queue_config(caplog):
+def test_queue_config(caplog, tmp_path):
     suiterc_content = """
 [scheduling]
    [[queues]]
@@ -231,12 +231,9 @@ def test_queue_config(caplog):
        inherit = A, B
    [[y]]
     """
-    workflow_directory = Path(mkdtemp())
-    suite_rc = Path(workflow_directory, "suite.rc")
-    with suite_rc.open(mode="w") as f:
-        f.write(suiterc_content)
-        f.flush()
-    config = SuiteConfig(suite="qtest", fpath=f.name)
+    suite_rc = tmp_path / "suite.rc"
+    suite_rc.write_text(suiterc_content)
+    config = SuiteConfig(suite="qtest", fpath=suite_rc.absolute())
     config.configure_queues()
     log = caplog.messages[0].split('\n')
     assert log[0] == "Queue configuration warnings:"
```

Reply:
(I learned recently about this `tmpdir`, probably will have to update some tests I wrote later :grimacing: )

Reply:
Done (good suggestion, thanks)